src/main/java/org/takes/facets/auth/Token.java

```diff
@@ -0,0 +1,187 @@
+         * JSON Object Signing and Encryption Header.
+         * @param bitlength Of encryption bits.
+         */
+        public Jose(final int bitlength) {
+            this.joseo = Json.createObjectBuilder()
+                .add(Jose.ALGORITHM, String.format("HS%s", bitlength))
+                .add(Jose.TYP, "JWT")
+                .build();
+        }
+
+        @Override
+        public JsonObject json() {
+            return this.joseo;
+        }
+
+        @Override
+        public byte[] encoded() {
+            byte[] bytes;
+            try {
+                bytes = new Base64().encode(this.joseo.toString());
+            } catch (final IOException ignore) {
+                bytes = new byte[] {};
+            }
+            return bytes;
+        }
+    };
+
+    /**
+     * JSON Web Token.
+     */
+    @SuppressWarnings
+        (
+            {
+                "PMD.ConstructorShouldDoInitialization",
+                "PMD.ConstructorOnlyInitializesOrCallOtherConstructors"
+            }
+        )
+    final class Jwt implements Token {
+        /**
+         * The header short for subject.
+         */
+        public static final String SUBJECT = "sub";
+
+        /**
+         * The header short for issuing time.
+         */
+        public static final String ISSUED = "iat";
+
+        /**
+         * The header short for expiration.
+         */
+        public static final String EXPIRATION = "exp";
+
+        /**
+         * The header short for expiration.
+         */
+        private static final String ISOFORMAT = "%tFT%<tRZ";
+
+        /**
+         * JWT object.
+         */
+        private final JsonObject jwto;
+
+        /**
+         * Time of lifespan start.
+         */
+        private final Calendar now = Calendar.getInstance(
+            TimeZone.getTimeZone("Z")
+        );
+
+        /**
+         * Time of lifespan end.
+         */
+        private final Calendar exp = Calendar.getInstance(
+            TimeZone.getTimeZone("Z")
+        );
+
+        /**
+         * JSON Web Token.
+         * @param idt Identity
+         * @param age Lifetime of token.
+         */
+        public Jwt(final Identity idt, final long age) {
+            // @checkstyle MagicNumber (1 line)
+            this.exp.setTimeInMillis(this.now.getTimeInMillis() + (age * 1000));
+            this.jwto = Json.createObjectBuilder()
+                .add(Jwt.ISSUED, String.format(Jwt.ISOFORMAT, this.now))
+                .add(Jwt.EXPIRATION, String.format(Jwt.ISOFORMAT, this.exp))
+                .add(Jwt.SUBJECT, idt.urn())
+                .build();
+        }
+
+        @Override
+        public JsonObject json() {
+            return this.jwto;
+        }
+
+        @Override
+        public byte[] encoded() {
+            byte[] bytes;
+            try {
+                bytes = new Base64().encode(this.jwto.toString());
+            } catch (final IOException ignore) {
+                bytes = new byte[] {};
```

Comment:
@semantosoph this looks a bit weird. Why so? Can't you just throw an unchecked exception here?